context/context.go

```diff
@@ -38,138 +32,136 @@ func ResolveRemotesToRepos(remotes Remotes, client *api.Client, base string) (Re
 		if err != nil {
 			return result, err
 		}
-		result.BaseOverride = baseOverride
+		result.baseOverride = baseOverride
 	}
 
-	foundBaseOverride := false
-	var hostname string
+	return result, nil
+}
+
+func resolveNetwork(result *ResolvedRemotes) error {
 	var repos []ghrepo.Interface
-	for i, r := range remotes {
-		if i == 0 {
-			hostname = r.RepoHost()
-		} else if !strings.EqualFold(r.RepoHost(), hostname) {
-			// ignore all remotes for a hostname different to that of the 1st remote
-			continue
-		}
+	for _, r := range result.remotes {
 		repos = append(repos, r)
-		if baseOverride != nil && ghrepo.IsSame(r, baseOverride) {
-			foundBaseOverride = true
-		}
 		if len(repos) == maxRemotesForLookup {
 			break
 		}
 	}
-	if baseOverride != nil && !foundBaseOverride {
-		// additionally, look up the explicitly specified base repo if it's not
-		// already covered by git remotes
-		repos = append(repos, baseOverride)
-	}
 
-	networkResult, err := api.RepoNetwork(client, repos)
-	if err != nil {
-		return result, err
-	}
-	result.Network = networkResult
-	return result, nil
+	networkResult, err := api.RepoNetwork(result.apiClient, repos)
+	result.network = &networkResult
+	return err
 }
 
 type ResolvedRemotes struct {
-	BaseOverride ghrepo.Interface
-	Remotes      Remotes
-	Network      api.RepoNetworkResult
+	baseOverride ghrepo.Interface
+	remotes      Remotes
+	network      *api.RepoNetworkResult
 	apiClient    *api.Client
 }
 
-// BaseRepo is the first found repository in the "upstream", "github", "origin"
-// git remote order, resolved to the parent repo if the git remote points to a fork
-func (r ResolvedRemotes) BaseRepo() (*api.Repository, error) {
-	if r.BaseOverride != nil {
-		for _, repo := range r.Network.Repositories {
-			if repo != nil && ghrepo.IsSame(repo, r.BaseOverride) {
-				return repo, nil
+func (r *ResolvedRemotes) BaseRepo(io *iostreams.IOStreams) (ghrepo.Interface, error) {
```

Comment:
What are your thoughts on changing this method to take in `canPrompt bool` argument instead of the `io *iostreams.IOStreams` argument? It seems that is all we are doing with `io` here, and might make the method easier to test.

Reply:
(fwiw, iostreams is easy to test)

Reply:
@samcoe That's an excellent point! It's usually better to pass primitives (such as boolean) than complex types such as IOStreams. Here, however, passing in IOStreams semantically draws attention that there will be prompt functionality inside. Right now, the Survey library isn't configured with IOStreams, but we are hoping to one day hook it up. After that, any part of functionality that needs access to stdin/stdout/stderr should take an IOStreams so it's easier to stub out in tests.